test_cluster_detection.py

```diff
@@ -0,0 +1,423 @@
+from timeit import default_timer as timer
+import sklearn.datasets
+from modules.cluster_estimation.cluster_estimation import ClusterEstimation
+from modules.cluster_estimation.detection_in_world import DetectionInWorld
+import numpy as np
+import pytest
+
+MIN_TOTAL_POINTS_THRESHOLD = 100
+MIN_NEW_POINTS_THRESHOLD = 10
+CENTER_BOX_SIZE = 500
+
+@pytest.fixture()
+def cluster_model():
+    model = ClusterEstimation()
+    yield model
+
+def cluster_data(n_samples:list, cluster_standard_deviation:int=1) -> list[DetectionInWorld]:
+    """
+    Returns a list of points (DetectionInWorld objects) with specified points per cluster
+    and standard deviation 
+    """
+    X, y = sklearn.datasets.make_blobs(n_samples=n_samples, 
+                                       n_features=2, 
+                                       cluster_std=cluster_standard_deviation, 
+                                       center_box=(0,CENTER_BOX_SIZE), 
+                                       random_state=0)
+    detections = []
+    for pair in X:
+        detections.append(DetectionInWorld(pair[0], pair[1]))
+    return detections, y
+
+class Test_Model_Execution_Condition():
+    """
+    Tests execution condition for estimation worker at different amount of total and new data points
+    """
+
+    def test_under_min_total_threshold(self, cluster_model):
+        """
+        Total data under threshold should not run 
+        """
+        # Setup 
+        NUM_DATA_POINTS = MIN_TOTAL_POINTS_THRESHOLD - 1  # less than min threshold (100)
+        X, y = cluster_data([NUM_DATA_POINTS])
+
+        # Run
+        model = cluster_model
+        model.clear_all_data()
```

Comment:
Do all tests succeed without if this call is completely removed?

Reply:
since the cluster_model object does not change, the pre-existing data would still be stored in cluster_mode.__all_points, causing the actual current # of data points in the model to be different than what is input each test. I can re-design the tests to work without this but it is more complicated. Let me know if I should do that instead.